src/main/java/org/owasp/html/HtmlEntities.java

```diff
@@ -28,13 +28,2129 @@
+    builder.put("dharr", Integer.valueOf('\u21c2')); // DOWNWARDS HARPOON WITH BARB RIGHTWARDS
+    builder.put("RightDownVector", Integer.valueOf('\u21c2')); // DOWNWARDS HARPOON WITH BARB RIGHTWARDS
+    builder.put("downharpoonright", Integer.valueOf('\u21c2')); // DOWNWARDS HARPOON WITH BARB RIGHTWARDS
+    builder.put("dharl", Integer.valueOf('\u21c3')); // DOWNWARDS HARPOON WITH BARB LEFTWARDS
+    builder.put("LeftDownVector", Integer.valueOf('\u21c3')); // DOWNWARDS HARPOON WITH BARB LEFTWARDS
+    builder.put("downharpoonleft", Integer.valueOf('\u21c3')); // DOWNWARDS HARPOON WITH BARB LEFTWARDS
+    builder.put("rlarr", Integer.valueOf('\u21c4')); // RIGHTWARDS ARROW OVER LEFTWARDS ARROW
+    builder.put("rightleftarrows", Integer.valueOf('\u21c4')); // RIGHTWARDS ARROW OVER LEFTWARDS ARROW
+    builder.put("RightArrowLeftArrow", Integer.valueOf('\u21c4')); // RIGHTWARDS ARROW OVER LEFTWARDS ARROW
+    builder.put("udarr", Integer.valueOf('\u21c5')); // UPWARDS ARROW LEFTWARDS OF DOWNWARDS ARROW
+    builder.put("UpArrowDownArrow", Integer.valueOf('\u21c5')); // UPWARDS ARROW LEFTWARDS OF DOWNWARDS ARROW
+    builder.put("lrarr", Integer.valueOf('\u21c6')); // LEFTWARDS ARROW OVER RIGHTWARDS ARROW
+    builder.put("leftrightarrows", Integer.valueOf('\u21c6')); // LEFTWARDS ARROW OVER RIGHTWARDS ARROW
+    builder.put("LeftArrowRightArrow", Integer.valueOf('\u21c6')); // LEFTWARDS ARROW OVER RIGHTWARDS ARROW
+    builder.put("llarr", Integer.valueOf('\u21c7')); // LEFTWARDS PAIRED ARROWS
+    builder.put("leftleftarrows", Integer.valueOf('\u21c7')); // LEFTWARDS PAIRED ARROWS
+    builder.put("uuarr", Integer.valueOf('\u21c8')); // UPWARDS PAIRED ARROWS
+    builder.put("upuparrows", Integer.valueOf('\u21c8')); // UPWARDS PAIRED ARROWS
+    builder.put("rrarr", Integer.valueOf('\u21c9')); // RIGHTWARDS PAIRED ARROWS
+    builder.put("rightrightarrows", Integer.valueOf('\u21c9')); // RIGHTWARDS PAIRED ARROWS
+    builder.put("ddarr", Integer.valueOf('\u21ca')); // DOWNWARDS PAIRED ARROWS
+    builder.put("downdownarrows", Integer.valueOf('\u21ca')); // DOWNWARDS PAIRED ARROWS
+    builder.put("lrhar", Integer.valueOf('\u21cb')); // LEFTWARDS HARPOON OVER RIGHTWARDS HARPOON
+    builder.put("ReverseEquilibrium", Integer.valueOf('\u21cb')); // LEFTWARDS HARPOON OVER RIGHTWARDS HARPOON
+    builder.put("leftrightharpoons", Integer.valueOf('\u21cb')); // LEFTWARDS HARPOON OVER RIGHTWARDS HARPOON
+    builder.put("rlhar", Integer.valueOf('\u21cc')); // RIGHTWARDS HARPOON OVER LEFTWARDS HARPOON
+    builder.put("rightleftharpoons", Integer.valueOf('\u21cc')); // RIGHTWARDS HARPOON OVER LEFTWARDS HARPOON
+    builder.put("Equilibrium", Integer.valueOf('\u21cc')); // RIGHTWARDS HARPOON OVER LEFTWARDS HARPOON
+    builder.put("nlArr", Integer.valueOf('\u21cd')); // LEFTWARDS DOUBLE ARROW WITH STROKE
+    builder.put("nLeftarrow", Integer.valueOf('\u21cd')); // LEFTWARDS DOUBLE ARROW WITH STROKE
+    builder.put("nhArr", Int
```

Comment:
Why not just use Integer.valueOf with the codepoint instead of doing an unnecessary conversion from surrogates?

Fyi, https://html.spec.whatwg.org/entities.json includes codepoints as well as the string form.

Reply:
Because Java characters are 16 bits and this codepoint requires the use of a surrogate to express in UTF-16, i.e. it cannot be expressed as a single Java character. Therefore we would have to use `Integer.valueOf(String)` which has a different behavior. Unless I'm misunderstanding your question.

Reply:
Why does it matter what Java characters are?  I just need an input I can use with `StringBuilder.appendCodepoint`.

Actually, I suppose for `"&NotEqualTilde;": { "codepoints": [8770, 824], "characters": "\u2242\u0338" },`  I really need a string that I can use with `StringBuilder.append`.

Perhaps this map should change to just store the replacement string which is just the stringified version of the chars but with backslash and double quotes escaped Java style.

Reply:
You're right, it doesn't matter. It could work either way and the output will be the same. This is executed once during class load and it's not clear that the performance will be noticeably different either way. Personally I think `\u2242` is more readable / traceable for the developer than `8770` but I think this is just an issue of code style. If you prefer it another way I can update it.

As you notice things are a bit tricker with references that refer to multiple code points. We could update `decodeEntityAt` to return a variable-length array. It turns out we can also stuff the code points into 32 bits, which is what I have now. The speculative idea being that it gives better performance because we don't have to allocate an array on the heap. But is it a noticeable difference? I don't know. We can do it either way.